src/qt/trafficgraphwidget.cpp

```diff
@@ -46,27 +63,75 @@ int TrafficGraphWidget::getGraphRangeMins() const
     return nMins;
 }
 
-void TrafficGraphWidget::paintPath(QPainterPath &path, QQueue<float> &samples)
+//MLUMIN 05_22: Resampling function neccessary for re-fitting all acquired data samples into displayed visual sample canvas/view.
+float TrafficGraphWidget::resampleSamples(const QVector<float> &source, QVector<float> &dest) const
 {
-    int h = height() - YMARGIN * 2, w = width() - XMARGIN * 2;
-    int sampleCount = samples.size(), x = XMARGIN + w, y;
-    if(sampleCount > 0) {
-        path.moveTo(x, YMARGIN + h);
-        for(int i = 0; i < sampleCount; ++i) {
-            x = XMARGIN + w - w * i / DESIRED_SAMPLES;
-            y = YMARGIN + h - (int)(h * samples.at(i) / fMax);
-            path.lineTo(x, y);
+    float stride = (float)nSamplesWindowSize / dest.size();
+    float numSamplesToCombine = stride;
+    float maxSample = 0.0f;
+    float lastSample = 0.0f;
+
+    size_t currentSampleIndex = nSamplesTailIndex;
+    for (size_t destIndex = 0; destIndex < dest.size(); destIndex++) {
+        int numSamplesToRead = (int)numSamplesToCombine;
+
+        float currentSample = 0.0f;
+
+        for (size_t i = 0; i < numSamplesToRead; ++i) {
```

Comment:
`i` should be an `int` because `numSamplesToRead` is an `int`.

Reply:
Same as two above, will bring into alignment with rest.